solutions/baseline/variables.tf

```diff
@@ -0,0 +1,286 @@
+  default     = null
+}
+
+variable "force_delete_storage" {
+  type        = bool
+  description = "Flag indicating whether or not to delete attached storage when destroying the cluster - Default: false"
+  default     = false
+}
+
+variable "number_worker_nodes" {
+  type        = number
+  description = "The number of workers to create in the cluster"
+  default     = 2
+}
+
+variable "machine_type" {
+  type        = string
+  description = "Worker node machine type. Use 'ibmcloud ks flavors --zone <zone>' to retrieve the list."
+  default     = "bx2.4x16"
+}
+
+variable "operating_system" {
+  type        = string
+  description = "Allowed OS values are RHEL_9 (RHEL_9_64), RHEL 8 (REDHAT_8_64) or Red Hat Enterprise Linux CoreOS (RHCOS). RHCOS requires VPC clusters created from 4.15 onwards. Upgraded clusters from 4.14 cannot use RHCOS."
+  default     = "RHEL_9_64"
+}
+
+variable "disable_outbound_traffic_protection" {
+  type        = bool
+  description = "Whether to allow public outbound access from the cluster workers. This is only applicable for `ocp_version` 4.15"
+  default     = true
+}
+
+variable "pod_subnet_cidr" {
+  type        = string
+  description = "Specify a custom subnet CIDR to provide private IP addresses for pods. The subnet must have a CIDR of at least `/23` or larger. Default value is `172.30.0.0/16` when the variable is set to `null`."
+  default     = null
+}
+
+variable "service_subnet_cidr" {
+  type        = string
+  description = "Specify a custom subnet CIDR to provide private IP addresses for services. The subnet must be at least `/24` or larger. Default value is `172.21.0.0/16` when the variable is set to `null`."
+  default     = null
+}
+
+variable "verify_worker_network_readiness" {
+  type        = bool
+  description = "By setting this to true, a script will run kubectl commands to verify that all worker nodes can communicate successfully with the master. If the runtime does not have access to the kube cluster to run kubectl commands, this should be set to false."
+  default     = true
+}
+
+variable "addons" {
+  type = object({
+    debug-tool                = optional(string)
+    image-key-synchronizer    = optional(string)
+    openshift-data-foundation = optional(string)
+    vpc-file-csi-driver       = optional(string)
+    static-route              = optional(string)
+    cluster-autoscaler        = optional(string)
+    vpc-block-csi-driver      = optional(string)
+    ibm-storage-operator      = optional(string)
+  })
+  description = "Map of OCP cluster add-on versions to install (NOTE: The 'vpc-block-csi-driver' add-on is installed by default for VPC clusters and 'ibm-storage-operator' is installed by default in OCP 4.15 and later, however you can explicitly specify it here if you wish to choose a later version than the default one). For full list of all supported add-ons and versions, see https://cloud.ibm.com/docs/containers?topic=containers-supported-cluster-addon-versions"
```

Comment:
This is complex input variable so it will need supporting markdown helper doc so projects consumers know the required format